controller/api/destination/endpoint_translator.go

```diff
@@ -383,29 +380,23 @@ func toWeightedAddr(address watcher.Address, opaquePorts, skippedInboundPorts ma
 	}, nil
 }
 
-func getK8sNodeTopology(nodes coreinformers.NodeInformer, srcNode string) (map[string]string, error) {
-	nodeTopology := make(map[string]string)
+func getNodeTopologyZone(nodes coreinformers.NodeInformer, srcNode string) (string, error) {
 	node, err := nodes.Lister().Get(srcNode)
 	if err != nil {
-		return nodeTopology, err
+		return "", err
 	}
-
 	for k, v := range node.Labels {
-		if k == corev1.LabelHostname ||
-			k == corev1.LabelZoneFailureDomainStable ||
-			k == corev1.LabelZoneRegionStable {
-			nodeTopology[k] = v
+		if k == corev1.LabelTopologyZone {
+			return v, nil
```

Comment:
Now that we only have one label that we're after perhaps we can just do a quicker lookup in the map rather than traverse it.

```suggestion
// take out 'for' loop
v, ok := node.Labels[corev1.LabelTopologyZone]
```